Undefined C library symbols when building Windows DLL

With zig 0.60 I can't build the example code below as a dynamic library. As an exe or static lib it builds fine though.

### Command line
```
zig.exe build-lib -dynamic app.zig -lc --release-safe
lld: error: undefined symbol: printf
>>> referenced by Q:\godot\native\src\app.zig:7
>>>               .\app.obj:(main)
```
### app.zig
```
const c = @cImport({
    @cDefine("_NO_CRT_STDIO_INLINE", "1");
    @cInclude("stdio.h");
});

pub export fn main() i32 {
    return c.printf("Hello!\n");
}
```

Guessing dlls linking libc have gone untested. With a completely empty main.zig, `zig.exe build-lib main.zig -dynamic -lc` gives me lld error: undefined symbol for:
- `__security_cookie`
- `__security_check_cookie`
- `memset`
- `memcpy`
- `__chkstk`

--- a/lib/std/start.zig
+++ b/lib/std/start.zig
@@ -54,7 +54,7 @@ fn _DllMainCRTStartup(
     fdwReason: std.os.windows.DWORD,
     lpReserved: std.os.windows.LPVOID,
 ) callconv(.Stdcall) std.os.windows.BOOL {
-    if (!builtin.single_threaded) {
+    if (!builtin.single_threaded and !builtin.link_libc) {
         _ = @import("start_windows_tls.zig");
     }
 

